lib/game.rb

```diff
@@ -0,0 +1,182 @@
+require 'pry'
+
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0,1,2], # top row
+    [3,4,5], # middle row
+    [6,7,8], # bottom row
+    [0,3,6], # left column
+    [1,4,7], # middle column
+    [2,5,8], # right column
+    [0,4,8], # left diagonal
+    [6,4,2] # right diagonal
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    self.board.turn_count.even? ? player_1 : player_2
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      if self.board.cells[combo[0]] == self.board.cells[combo[1]] && self.board.cells[combo[1]] == self.board.cells[combo[2]] && self.board.cells[combo[0]] != " "
+        combo
+      end
+    end
+  end
+
+  def draw?
+    !self.won? && self.board.full?
+  end
+
+  def over?
+    self.won? || self.draw?
+  end
+
+  def winner
+    if self.won?
+      self.board.cells[self.won?[0]]
+    end
+  end
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.